core/common/src/Clock.kt

```diff
@@ -35,3 +35,17 @@ private class InstantTimeMark(private val instant: Instant, private val clock: C
 
     override fun minus(duration: Duration): TimeMark = InstantTimeMark(instant - duration, clock)
 }
+
+/**
+ * Returns the [Clock] by storing an initial [TimeMark] using [TimeSource.markNow] and [returns][Clock.now] the elapsed
+ * time using [TimeMark.elapsedNow] plus the provided [offset].
+ *
+ * This clock stores the initial [TimeMark], so repeatedly creating [Clock]s from the same [TimeSource] results
+ * into different [Instant]s iff the time of the [TimeSource] was increased. To sync different [Clock]s, use the [offset]
+ * parameter.
+ */
+@ExperimentalTime
+public fun TimeSource.toClock(offset: Instant = Instant.fromEpochSeconds(0)): Clock = object : Clock {
```

Comment:
Have you considered some other default value for `offset`, e.g. `Clock.System.now()`, or no default value at all?

Reply:
I did. The main use-case is converting the testTimeSource of coroutines to a Clock for high-level datetime usage, like Instant, with coroutines sync support, like `delay`. I prefer deterministic values for easier testing, like assertEquals or using test fixtures. With this use-case, I wouldn't use `Clock.System.now()` using the current changing time.
I am open to having no default value, but this would require passing an offset each time using this converter and if you want to customize the offset, you can already do it by passing your offset explicitly. So I don't see a reason to remove a convenience default value.
I used the start of the epoch, because, well, it is the start and it's commonly used for test fixtures.